weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTable.tsx

```diff
@@ -676,6 +684,25 @@ export const CallsTable: FC<{
   }, [allRowKeys, columnVisibilityModel, tableData]);
 
   const [deleteConfirmModalOpen, setDeleteConfirmModalOpen] = useState(false);
+  const [addToDatasetModalOpen, setAddToDatasetModalOpen] = useState(false);
+  const [selectedCallObjects, setSelectedCallObjects] = useState<CallData[]>(
+    []
+  );
+
+  // Update selected call objects whenever rows data changes
+  useEffect(() => {
```

Comment:
This use effect is causing additional re-render. i think a simple memo would suffice